installer/PowerToysSetupCustomActions/CustomAction.cpp

```diff
@@ -38,6 +39,53 @@ const DWORD USERNAME_LEN = UNLEN + 1; // User Name + '\0'
 static const wchar_t* POWERTOYS_EXE_COMPONENT = L"{A2C66D91-3485-4D00-B04D-91844E6B345B}";
 static const wchar_t* POWERTOYS_UPGRADE_CODE = L"{42B84BF7-5FBF-473B-9C8B-049DC16F7708}";
 
+constexpr inline const wchar_t* DataDiagnosticsRegKey = L"Software\\Classes\\PowerToys";
+constexpr inline const wchar_t* DataDiagnosticsRegValueName = L"AllowDataDiagnostics";
+
+#define TraceLoggingWriteWrapper(provider, eventName, ...)   \
+    if (isDataDiagnosticEnabled())                           \
+    {                                                        \
+        trace.UpdateState(true);                             \
+        TraceLoggingWrite(provider, eventName, __VA_ARGS__); \
+        trace.Flush();                                       \
+        trace.UpdateState(false);                            \
+    }
+
+static Shared::Trace::ETWTrace trace{ L"PowerToys_Installer" };
+
+inline bool isDataDiagnosticEnabled()
```

Comment:
Should respect windows policy or PowerToxs policy too. Or am I wrong?

Reply:
only powertoys